lamindb/models/artifact_set.py

```diff
@@ -40,6 +46,47 @@ class ArtifactSet(Iterable):
         >>> artifacts # an instance of ArtifactQuerySet inheriting from ArtifactSet
     """
 
+    def _filter_with_features(self, *queries, **expressions) -> ArtifactSet:
```

Comment:
One issue with this is that `Record` and `Run` relates to features in the same way as `Artifact`.

All of `Artifact`, `Record`, and `Run` should be queryable features with the same mechanism. In fact, `ln.Record.to_dataframe(features=True)` already works in the same way as `ln.Artifact.to_dataframe(features=True)`, creating a table in which features map into columns of that table.

But this just for context for where the implementation should go in the future.